app/models/user.rb

```diff
@@ -73,7 +73,10 @@ def self.recent_orders_created_for(user_id)
   end
 
   def self.search(searchText, role)
-    joins(:roles).where("roles.name = ?  AND first_name ILIKE ? OR last_name ILIKE ?", role, "%#{searchText}%", "%#{searchText}%").uniq
```

Comment:
I've also written a very similar method (still in progress) for `offer.search` on another branch. https://github.com/crossroads/api.goodcity/blob/offers-shallow-load/app/models/concerns/offer_search.rb

 Can you do a similar approach, particularly:
* handling case where searchText is empty
* moving into a model concern
* use LEFT OUTER JOIN (I think this avoids `uniq`?)
* Fix bug for AND and OR (unless I'm missing something here?)